src/index.ts

```diff
@@ -147,71 +149,91 @@ function getRemoteMetadata(ctx) {
     let ret;
 
     if (
-      ctx._oembed.type === "application/json+oembed" &&
+      oembed.type === "application/json+oembed" &&
       /application\/json/.test(contentType)
     ) {
       ret = await res.json();
     } else if (
-      ctx._oembed.type === "text/xml+oembed" &&
+      oembed.type === "text/xml+oembed" &&
       /(text|application)\/xml/.test(contentType)
     ) {
-      const data = await res.text();
+      const data = (await res.text())
+        .replace(/&gt;/g, ">")
+        .replace(/&lt;/g, "<");
 
       const content: { [key: string]: string } = {};
 
+      const parserContext: ParserContext = { text: "" };
+
       ret = await new Promise((resolve) => {
-        const parser = new Parser({
-          onopentag: function (name, attribs) {
-            if (this._is_html) {
-              if (!content.html) {
-                content.html = "";
+        const parser = new Parser(
+          {
+            oncdataend: () => {
```

Comment:
Now we pass `recognizeCDATA` we can add a `oncdataend` handler which will get called when the ctag closes.